include/ocpp/v201/smart_charging.hpp

```diff
@@ -48,29 +54,45 @@ std::ostream& operator<<(std::ostream& os, const ProfileValidationResultEnum val
 class SmartChargingHandler {
 private:
     std::map<int32_t, std::unique_ptr<EvseInterface>>& evses;
+    std::unique_ptr<DeviceModel>& device_model;
 
     std::shared_ptr<ocpp::v201::DatabaseHandler> database_handler;
     // cppcheck-suppress unusedStructMember
     std::map<int32_t, std::vector<ChargingProfile>> charging_profiles;
     std::vector<ChargingProfile> station_wide_charging_profiles;
 
 public:
-    explicit SmartChargingHandler(std::map<int32_t, std::unique_ptr<EvseInterface>>& evses);
+    explicit SmartChargingHandler(std::map<int32_t, std::unique_ptr<EvseInterface>>& evses,
+                                  std::unique_ptr<DeviceModel>& device_model);
+
+    ///
+    /// \brief validates the given \p profile according to the specification.
+    /// If a profile does not have validFrom or validTo set, we conform the values
+    /// to a representation that fits the spec.
+    ///
+    ProfileValidationResultEnum validate_profile(ChargingProfile& profile, int32_t evse_id);
```

Comment:
If we do that, we feel like we would lose the ability to easily unit test and lose the high level of coverage that we currently have. It would also make it difficult to test-drive functionality that remains to be implemented. Would it be acceptable to make them `protected` instead of `private`?

Reply:
This was brought up on the General Q&A call and it does not appear that the community has any concern with `protected` methods.

Reply:
If that is the best option then I have no problem with that. Just felt strange having them exposed while the user of the class should not use them.

Reply:
@marcemmers it is a reasonable request. We just want to balance the ability to test and locking the interface down.

Reply:
Marked them all as protected